airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/adapters.py

```diff
@@ -174,7 +174,18 @@ def read_records(
         stream_slice: Optional[Mapping[str, Any]] = None,
         stream_state: Optional[Mapping[str, Any]] = None,
     ) -> Iterable[StreamData]:
-        yield from self._read_records()
+        try:
```

Comment:
It seems like this relies on specific implementation/private parameters. It means that I might be developing in a Cursor, change _state to be something else and break this error log. Should we make it explicit as part of the cursor interface?

Reply:
Good point. Made `state` public on the cursor.